src/AnimalFamily/AnimalFamily.ts

```diff
@@ -1,30 +1,32 @@
+import Crocodile from "../animals/Crocodile";
+import Lion from "../animals/Lion";
 import { families } from "../app";
-import {
-  AnimalType,
-  MammalAnimalType,
-  ReptileAnimalType,
-} from "../customTypes";
+import { AnimalType } from "../customTypes";
 import verifyFamilyInfo, {
   VerificationResult,
   VerificationProps,
 } from "./verifyFamilyInfo";
 export default class AnimalFamily {
   name: string;
   animals: AnimalType[];
-  animalsType: MammalAnimalType | ReptileAnimalType;
+  animalsType: typeof Lion | typeof Crocodile;
```

Comment:
I am not sure if its okay like this.
It works just not sure if writing it like this is good or if i should use any other way.

Reply:
Why do you even need to store the animals type? 
Whenever you need it just check what instance is animal[0]  ( if there are animals in it) like you dud a couple of lines lower